src/zarr/testing/strategies.py

```diff
@@ -209,7 +210,7 @@ def basic_indices(draw: st.DrawFn, *, shape: tuple[int], **kwargs: Any) -> Any:
 
 
 def key_ranges(
-    keys: SearchStrategy = node_names, max_size: int | None = None
+    keys: SearchStrategy = node_names, max_size: int = sys.maxsize
```

Comment:
This is [part of our public API](https://zarr--2804.org.readthedocs.build/en/2804/api/zarr/testing/strategies/index.html#zarr.testing.strategies.key_ranges), so changing the default here is an API change. What was the motivation for changing it - does hypothesis not work well with `max_value=None`?

Reply:
Cleans up the use of `min` below, that's all.

Reply:
👍